storm-core/src/jvm/org/apache/storm/windowing/CountEvictionPolicy.java

```diff
@@ -26,10 +26,10 @@
  * @param <T> the type of event tracked by this policy.
  */
 public class CountEvictionPolicy<T> implements EvictionPolicy<T> {
-    private final int threshold;
+    private final long threshold;
```

Comment:
For counts int should be sufficient. 

If you are changing it to long, the `currentCount` below should be changed to AtomicLong. You should also change the definition of `Count` class and add overloaded methods to take a long value.